### ISSUE TITLE
question about training phi

### ISSUE THREAD

**Issue by Juice113** [miccunifi/SEARLE#8]
Thank you for sharing this work! 
I have a question about the training process(in Figure 2) when reading the paper; I wonder is it: firstly we use loss "Lcos and Lgpt" to train the OTI(left part) ; then we train network phi using the trained OTI(the red cylinder) and Lgpt(right part). Is it right?

another question about the  pseudo-word S*: I find that S* was inserted into the relative caption(in figure 1). Is the S* a readable word like "dog" or something else?

**Comment by LorenzoAgnolucci** [comment]
Hi @Juice113

Thanks for your interest in our work!

Regarding your first question, the answer is yes: in the OTI phase, we generate the pseudo-word tokens related to the pre-training dataset using Lcos and Lgpt. Then, we train phi by distilling the knowledge of the OTI-generated tokens via Ldistil and Lgpt.

Regarding your second question, S* does not really represent any readable actual word (thus the name "pseudo-word"), it is just a mapping of the input image in the CLIP token embedding space.

Feel free to ask other questions and please star our repo if you find our work useful.

**Comment by Juice113** [comment]
for the pseudo-word S*: then how to "concatenate" it with the relative caption? I find that the full sentence "a photo of S* that is playing with a red ball" was input to CLIP text encoder in Figure 1.

**Comment by LorenzoAgnolucci** [comment]
Please take a look at [this](https://github.com/miccunifi/SEARLE/blob/8de1a6652814cabc85aabdeae9201a6db528c94a/src/encode_with_pseudo_tokens.py#L5).

**Comment by Juice113** [comment]
thank you.
